nwb_conversion_tools/utils/si013nwbephyswriter.py

```diff
@@ -0,0 +1,302 @@
+    stub_channels: list
+        channels to include when writing as stub
+    """
+
+    def __init__(self, object_to_write, stub=False, stub_channels=None):
+        assert HAVE_SI013, "spikeextractors 0.13 version not installed"
+        BaseSINwbEphysWriter.__init__(self, object_to_write, stub=stub, stub_channels=stub_channels)
+        if isinstance(self.object_to_write, se.RecordingExtractor):
+            self.recording = self.object_to_write
+            if self.stub:
+                self._make_recording_stub()
+        elif isinstance(self.object_to_write, se.SortingExtractor):
+            self.sorting = self.object_to_write
+            if self.stub:
+                self._make_sorting_stub()
+
+    def _make_recording_stub(self):
+        if self.stub_channels is not None:
+            channel_stub = self.stub_channels
+        else:
+            num_channels = min(10, self.recording.get_num_channels())
+            channel_stub = self.recording.get_channel_ids()[:num_channels]
+        frame_stub = min(100, self.recording.get_num_frames())
+        self.recording = se.SubRecordingExtractor(self.recording, channel_ids=channel_stub, end_frame=frame_stub)
+
+    def _make_sorting_stub(self):
+        max_min_spike_time = max(
+            [min(x) for y in self.sorting.get_unit_ids() for x in [self.sorting.get_unit_spike_train(y)] if any(x)]
+        )
+        self.sorting = se.SubSortingExtractor(self.sorting, start_frame=0, end_frame=1.1 * max_min_spike_time)
+
+    @staticmethod
+    def supported_types():
+        assert HAVE_SI013
+        return (se.RecordingExtractor, se.SortingExtractor)
+
+    def get_num_segments(self):
+        return 1
+
+    def _get_traces(self, channel_ids=None, start_frame=None, end_frame=None, return_scaled=True, segment_index=0):
+        return self.recording.get_traces(
+            channel_ids=channel_ids, start_frame=start_frame, end_frame=end_frame, return_scaled=return_scaled
+        ).T
+
+    def _get_dtype(self, return_scaled=True):
+        return self.recording.get_dtype(return_scaled)
+
+    def _get_channel_property_names(self):
+        property_names = set()
+        for chan_id in self._get_channel_ids():
+            for i in self.recording.get_channel_property_names(chan_id):
+                property_names.add(i)
+        return list(property_names)
+
+    def _fill_missing_property_values(self, ids, prop, get_prop_func):
+        self.dt_column_defaults = {list: [], str: "", Real: np.nan, np.ndarray: np.array([np.nan])}
+        # find the size of ndarray dtype:
+        for id in ids:
+            try:
+                id_data = get_prop_func(id, prop)
+                if isinstance(id_data, np.ndarray):
+                    self.dt_column_defaults.update({np.ndarray: np.nan * np.ones(shape=[1, id_data.shape[1:]])})
+                    break
+                else:
+                    break
+            except:
+                continue
```

Comment:
Why do we have these except clauses? Is there a better way to write this?

Reply:
changed as discussed https://github.com/catalystneuro/nwb-conversion-tools/pull/230/commits/c26593605361bc92cafb5305b9096f51e6145d5e